src/assignment-cache/local-storage-assignment-cache.ts

```diff
@@ -1,31 +1,17 @@
 import { AssignmentCache } from '@eppo/js-client-sdk-common';
 
-import { hasWindowLocalStorage } from './configuration-factory';
-
 class LocalStorageAssignmentShim {
   LOCAL_STORAGE_KEY = 'EPPO_LOCAL_STORAGE_ASSIGNMENT_CACHE';
 
   public has(key: string): boolean {
-    if (!hasWindowLocalStorage()) {
-      return false;
-    }
```

Comment:
IMO no need for these methods - the class should only be instantiated from the factory and when `localStorage` is detected. If not, throwing an exception is "good" behavior so the end-user uses the more appropriate class.